modified original version to write notes to csv file and to html file

mjb101 opened the pull request:
Have modified original code as below.
Output files are written to same directory as script.
Works on my backup .doc files from 2015-2022.
All seems to work OK locally - Windows 10 Home, Python 3.10.8.
[Tried using 'tabulate' for html which works well, except output file bloated by spaces (eg 16MB vs 300KB)]

decode-ColorNote-CSV.py
modified to write all notes to a csv file
fields written may be selected by editing json_keys_select

decode-ColorNote-CSV-HTML.py
decode-ColorNote-CSV-HTML-TEMPLATE.html
modified handling of json_keys_select
further modified to write all notes to an html table
table rows are color coded as per ColorNote
table may be sorted by clicking on column headers
requires html template file to be in same directory as python script
requires that "color_index" be included in json_keys_select

mjb101 committed:
Update decode-ColorNote.py

changed tabs to spaces
modified path creation for "notes.bin"
added offset to variable idx to allow for different doc formats

mjb101 committed:
Add files via upload

modified to write all notes to a csv file
fields written may be selected by editing json_keys_select

mjb101 committed:
Add files via upload

modified to write all notes to an html table
table rows are color coded as per ColorNote
table may be sorted by clicking on column headers
requires html template file to be in same directory as python script
requires that "color_index" be included in json_keys_select

mjb101 committed:
Add files via upload

modified to handle '.dat' files (in addition to '.doc' files) by changing magic_offset
modified file handling for backup, tmp and output files
renamed 'decoded_doc' as 'decoded_backup_file'
various minor tweaks

mjb101 committed:
Update decode-ColorNote-CSV-HTML.py

modified to handle '.backup' files
added import sys

fcoiffie commented:
Thank you for your code. I'm trying to merge the 3 scripts in only one with several arguments to define the output (like --csv or --html). I began in my own branch with merging the old one and your CSV version. But I still have to merge the HTML version.

mjb101 commented:
You're welcome. Thanks for the update. Look forward to seeing the new version.
________________________________
From: Frédéric Coiffier ***@***.***>
Sent: 05 December 2022 15:24
To: fcoiffie/decode-ColorNote ***@***.***>
Cc: mjb101 ***@***.***>; Author ***@***.***>
Subject: Re: [fcoiffie/decode-ColorNote] modified original version to write notes to csv file and to html file (PR #3)


Thank you for your code. I'm trying to merge the 3 scripts in only one with several arguments to define the output (like --csv or --html). I began in my own branch with merging the old one and your CSV version. But I still have to merge the HTML version.

—
Reply to this email directly, view it on GitHub<https://github.com/fcoiffie/decode-ColorNote/pull/3#issuecomment-1337569329>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AKGCEUBEDJSUVONRJ3WH2W3WLYCJ3ANCNFSM6AAAAAASIURGVE>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

mjb101 committed:
macros to highlight rows as per ColorNote colors

apply to '.csv' file opened with Microsoft Excel or LibreOffice Calc
for usage, read instructions within each macro file

mjb101 committed:
Update decode-ColorNote-CSV-HTML-CALC-MACRO_ColorNoteRows.bas

changed tabs to spaces

mjb101 committed:
Update decode-ColorNote-CSV-HTML.py

code correction to define 'json_keys' as a list for case when 'json_keys_select' is specified as an empty list

mjb101 committed:
Update decode-ColorNote-CSV-HTML.py

modified to correct handling of '.dat' files by fixing value of 'idx'
various minor tweaks
